### ISSUE TITLE
Considering (partial?) removal of openalias support. Comment here if you use it.

### ISSUE THREAD

**Issue by SomberNight** [spesmilo/electrum#6232]
With recent refactoring of invoices/payment requests, OpenAlias support is now partially broken.
Looking up an OpenAlias (name -> addr) works, however
- signing a payment request with an openalias is broken
  https://github.com/spesmilo/electrum/blob/eb39aa143b8112549a5f8156d8e3ee41034eac6a/electrum/wallet.py#L1772
- trying to pay a bip21 URI with embedded openalias name and signature is broken
  https://github.com/spesmilo/electrum/blob/eb39aa143b8112549a5f8156d8e3ee41034eac6a/electrum/util.py#L854
  https://github.com/spesmilo/electrum/blob/eb39aa143b8112549a5f8156d8e3ee41034eac6a/electrum/paymentrequest.py#L449

I am wondering whether this functionality is worth restoring/fixing.
We could remove the broken functionality;
or maybe we could strip OpenAlias support completely.

Do people actually use OpenAlias?
Do you use OpenAlias? What do you use it for?

If you want the project to keep supporting OpenAlias, please comment here.

**Comment by fluffypony** [comment]
I know of a fair number of people that use OpenAlias within Electrum, and we're actually busy with an OpenAlias v2 spec. There's also some activity around OpenAlias with Lightning, so I'd definitely push for it to continue.

A bip21 URI with OA embedded could get confusing, as the OA record itself could include an amount, so I'd veer away from that particular edge-case - unless we start to see adoption by payment processors that dictates revisiting this.

**Comment by relativisticelectron** [comment]
Without paynyms is this not a huge privacy leak and should not be used?

**Comment by fluffypony** [comment]
@relativisticelectron yes, but that doesn't mean it's not useful. OpenAlias is a very broad standard, and works with Monero (for instance) where privacy isn't an issue. It already allows you to specify a Lightning invoice (via ```oa1:lightning```), which may be somewhat useless given the nature of DNS, but there will be long-lived LN payment mechanisms in future that OA will support.

**Comment by ysangkok** [comment]
@fluffypony keysend is already supported in LND and c-lightning: https://lightning.readthedocs.io/lightning-keysend.7.html

**Comment by fluffypony** [comment]
@ysangkok awesome that's great news!

**Comment by JeremyRand** [comment]
Namecoin is interested in OpenAlias (specifically using `.bit` domains for OpenAlias records), so we'd definitely vote for keeping the OpenAlias support present.  Definitely would be useful to fix the privacy leaks associated with address reuse though.

**Comment by losnappas** [comment]
Who is this built for? Is there even a UI for registering an alias? I think it's junk, have not seen anybody use it.

**Comment by fluffypony** [comment]
@losnappas when you say stuff like "is there a UI for registering an alias" you demonstrate that you have no clue what you're talking about, didn't bother reading up on it, and are merely giving your uninformed opinion for some narcissistic reason. Feel free to come back to this conversation when you've actually done a smattering of research.

**Comment by losnappas** [comment]
Whoa there.

I was just giving the 2 cents of the average user. "for some narcissistic reason"? Calm down.

**Comment by fluffypony** [comment]
> I was just giving the 2 cents of the average user. "for some narcissistic reason"? Calm down.

If that was the case you would have just said "I don't personally know anyone using it". Comments like "it's junk" have no place here.

**Comment by losnappas** [comment]
Right, I thought that that was more of a funny thing to say, but I guess people will read it with another tone in mind.

My apologies

**Comment by dunnock** [comment]
This really helps to send coins to people I personally know, instead of keeping hexadecimal addresses at hand for copy pasting. Please don't remove.

**Comment by delta1** [comment]
Another vote to please keep OpenAlias support, I use it regularly and I'm certain many more people (that may never see this issue do as well).

**Comment by mikethetike** [comment]
I'll like the project to keep supporting OpenAlias

**Comment by Gangster8192** [comment]
How about making OpenAlias one of those default plugins that you can enable?

**Comment by artk47** [comment]
Is there any further activity re OA v2 spec? The site looks doomed..
